compiler/src/dmd/mars.d

```diff
@@ -1937,10 +1944,38 @@ bool createModules(ref Strings files, ref Strings libmodules, ref Param params,
         modules.push(m);
         if (firstmodule)
         {
-            global.params.objfiles.push(m.objfile.toChars());
+            params.objfiles.push(m.objfile.toChars());
             firstmodule = false;
         }
     }
+
+    // Special module representing `stdin`
+    if (params.readStdin)
+    {
+        Module m;
+        if (createModule("__stdin.d", libmodules, params, target, eSink, m))
+            return true;
+        if (m is null)
+            return false;
+
+        modules.push(m);
+
+        // Set the source file contents of the module
+        OutBuffer buf;
+        buf.readFromStdin();
+        m.src = cast(ubyte[])buf.extractSlice();
+
+        // Give unique outfile name
+        OutBuffer namebuf;
+        namebuf.printf("__stdin_%d", getpid());
```

Comment:
Not in scope of this PR since it was already there, but what's the point of mangling pid in the output file?